src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    private void grow() {
+        Object[] newArray = new Object[array.length * 3 / 2];
+        System.arraycopy(array, 0, newArray, 0, array.length);
+        array = (T[]) newArray;
+
+    }
 
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            if (size == array.length) {
```

Comment:
you doing this check on both cases. I would suggest to move it to the beginning of the method, before conditions

Reply:
done